gb-clock/src/debuger.rs

```diff
@@ -0,0 +1,7 @@
+use gb_bus::Bus;
+
+/// The interface for the clock's usage of the debuger.
+pub trait Debuger<B: Bus<u8> + Bus<u16>> {
```

Comment:
1. this trait will be implemented by whom ? 
2. how the Bus can help to know if a break point is hit ?

Reply:
1. It should be implemented by the Debuger from `gb-dbg`, or any entity that can detect if a breakpoint has been reached.
2. From my understanding we need the pc and the list of instruction to detect if we've reached a breakpoint. I thought we could get these from the bus, but perhaps we should give the cpu instead ?

Maybe its too early to add the debuger in the frame execution ?

Reply:
the `PC` is a register of the `CPU`.
I expect from a `Bus` to only mapped area of memory together.

I think it's a bit too early to have in mind the debuger.
I was thinking when the debuging is enabled, it's in charge of doing the `clock` (by controlling the clock or something else)

Reply:
We could pass the cpu as well, but I don't think it is possible yet, at least until it implement ticker. 

What should I do ? The `Debuger` is optional for the `frame` function from the `Clock` so it won't block us. Perhaps we can merge this PR and work on this trait later ? Or I can remove it for now.

Reply:
for the time being remove the trait limitation to `Bus`

Reply:
Done, I've anonymized the variable, we'll see later what is necessary for this function.